Feature/mashup-test-button

tirirote opened the pull request:
Closes #58

alvarobernal2412 committed:
feat: added draft catalog step

alvarobernal2412 committed:
feat(dashboard): added empty dashboard form

alvarobernal2412 committed:
Merge branch 'multi-step-form' into enhance-dashboard-module

alvarobernal2412 committed:
feat(catalog): enhanced controls creation step

alvarobernal2412 committed:
feat(panel): added panel creation form

alvarobernal2412 committed:
feat(multi-step): added catalog recovery

ferferga committed:
refactor: rewrite calendar component for react-day-picker v9

shadcn hasn't updated their component for react-day-picker v9 yet, it's still on v8.
However, in the styling docs of React Day Picker (https://daypicker.dev/docs/styling), they provide an unoffocial sample file (https://date-picker.luca-felix.com/)

This is where this new version has been sourced, but all Typescript syntax has been stripped

Signed-off-by: Fernando Fernández <ferferga@hotmail.com>

alvarobernal2412 committed:
Merge remote-tracking branch 'origin/fix-reactdarypicker-v9-shadcn' into enhance-dashboard-module

alvarobernal2412 committed:
Merge remote-tracking branch 'origin' into enhance-dashboard-module

alvarobernal2412 committed:
Merge branch 'main' into enhance-dashboard-module

alvarobernal2412 committed:
feat(catalog): added edit functionality

alvarobernal2412 committed:
refactor(dashboard): simplified panel model

alvarobernal2412 committed:
feat(panel): enhanced SQL query builder and form UI

alvarobernal2412 committed:
feat(panel): improved panel preview functionality

alvarobernal2412 committed:
feat(dashboard): added new panel types and restructure options

alvarobernal2412 committed:
refactor(dashboard): clean up code and improve type handling

alvarobernal2412 committed:
feat(dashboard): simplify dashboard configuration UI

alvarobernal2412 committed:
feat(dashboard): integrated panel creation in multi-step-form

alvarobernal2412 committed:
feat: improve dashboard draft handling

alvarobernal2412 committed:
fix: enhance control error handling

alvarobernal2412 committed:
fix: improve panel handling

alvarobernal2412 committed:
Merge branch 'main' into enhance-dashboard-module

renovate[bot] committed:
chore(deps): update npm (development)

tirirote committed:
feat: added wip mashup testing button

tirirote committed:
feat: added wip mashup testing button

tirirote committed:
Merge branch 'feature/mashup-test-button'  into feature/mashup-test-button

tirirote committed:
feat: added full wip testing button

alvarobernal2412 commented on src/services/mashups.js:
```diff
@@ -64,4 +111,118 @@ export function getFlowParams(flowId) {
+ * @param {object} payload - El objeto 'msg' a enviar al flujo (ej. { payload: 'Hello' }).
+ * @returns {Promise<any>} - La respuesta directa del flujo de Node-RED.
+ */
+export async function executeNodeRedMashup(url, payload) {
+  const username = 'admin';
+  const password = 'admin123';
+
+  if (!username || !password) {
+    console.warn("Advertencia: Las credenciales de Node-RED no están configuradas correctamente. La autenticación podría fallar.");
+    // Considera lanzar un error o manejar esto de forma más robusta en producción.
+  }
+
+  // 2. Codificar las credenciales en Base64
+  const credentials = encodeToBase64(`${username}:${password}`);
+
+  try {
+    const response = await nodeRedClient.post(`/api/v1/${url}`, payload, {
+      headers: {
+        'Content-Type': 'application/json',
+        // 3. Añadir el encabezado Authorization
+        'Authorization': `Basic ${credentials}`
+      }
+    });
+    return response.data;
+  } catch (error) {
+    console.error('Error executing Node-RED mashup:', error.response ? error.response.data : error.message);
+    throw error; // Re-lanza el error para que sea manejado por el componente de UI
+  }
```
We have the nodeRedToken in the cookies so the `username` and `password` are not necessary.

alvarobernal2412 commented on src/services/mashups.js:
```diff
@@ -64,4 +111,118 @@ export function getFlowParams(flowId) {
+  }
+}
+
+/**
+ * Crea un control de prueba en el backend.
+ * Ajustado para devolver directamente el objeto de respuesta del API,
+ * asumiendo que el control ya está en la raíz de la respuesta HTTP.
+ *
+ * @param {string} mashupName - El nombre del mashup para el que se está creando el control de prueba.
+ * @param {string} selectedCatalogId - El ID del catálogo seleccionado por el usuario en el modal.
+ * @returns {Promise<object>} - Promesa que resuelve con los datos del control creado.
+ */
```
This should be in English

alvarobernal2412 commented on src/services/mashups.js:
```diff
@@ -1,38 +1,86 @@
+      return tab;
     });
+
+    console.log("Flujos de Node-RED procesados:", allFlowsInfo); // Para depuración
```
This should be removed

alvarobernal2412 commented on src/services/mashups.js:
```diff
@@ -1,38 +1,86 @@
+
+  } catch (error) {
+    console.error("Error fetching all Node-RED flows:", error);
+    // Relanzar el error para que pueda ser capturado por el código que llama
```
This should be in English

alvarobernal2412 commented on src/pages/app/Mashups.jsx:
```diff
@@ -62,7 +71,9 @@ export function Mashups() {
       setLoading(true);
-      const response = await getAllApiFlows();
+      const response = await getAllNodeRedFlows();
+      // Asumimos que getAllNodeRedFlows devuelve también 'mainInputType' y 'url'
+      // Si no es así, deberías ajustar la llamada o el servicio getAllNodeRedFlows
```
This should be in English

alvarobernal2412 commented on src/components/mashups/ControlCreationAndTestView.jsx:
This file is a mix between a component and a page, a component must be reusable and a page must include only its features. E.g. The Mashup Page must only include the fetch, see (redirect to Details page), test mashup (redirect to Test Mashup Page) and delete features.

alvarobernal2412 commented on src/forms/control/new/form.jsx:
Be careful with this type of changes because this form is already used for other features, if you need to change its logic, then create other form. If the logic you're introducing doesn't break other uses, then reuse this form.

alvarobernal2412 reviewed (changes requested):
The mashup response include is given by `/computations/id` where id is then `computationGroup`. Response example:
```json
[
    {
      id: computationGroup,
      value: complianceValue,
      scope: trimmedScope,
      evidences: computations,
      period,
      controlId,
    },
]
```

Each computation in evidence attribute has this structure:
```text
 id: d290f1ee-6c54-4b01-90e6-d701748f0851
 computationGroup: 123e4567-e89b-12d3-a456-426614174000
 value: true
 scope: {"project": "showcase-GH-governify_bluejay-showcase", "class": "showcase", "member": "Javi_Fdez" }
 evidences: [{"document": "Document confirming the computation"}]
 period: { "from": "2022-04-07T02:00:00.000Z", "to": "2022-04-07T02:59:59.999Z" }
 controlId: 2
```


You must take that into account for the mashup test response. Cheer up!

alvarobernal2412 committed:
wip(panels): added panel card

tirirote committed:
feat: added wip changes for the testing button

alvarobernal2412 committed:
feat: replace modal mashup testing with dedicated page

alvarobernal2412 committed:
Merge branch 'enhance-dashboard-module' into feature/mashup-test-button

alvarobernal2412 committed:
feat: add IndexedDB utility for mashup test persistence

- Add comprehensive IndexedDB manager class with CRUD operations
- Support test storage by mashup ID with timestamp indexing
- Include bulk operations for clearing tests
- Provide singleton instance for app-wide usage

alvarobernal2412 committed:
feat: add custom hook for mashup test management

- Create useMashupTests hook for centralized test state
- Integrate with IndexedDB for persistent storage
- Provide both async and sync access patterns
- Support CRUD operations with local state optimization

alvarobernal2412 committed:
feat: add reusable components for test results display

- Add PreviousTests component for test history management
- Add TestResults component for current test output display
- Include actions for viewing, deleting, and managing tests
- Support formatted timestamps and result visualization

alvarobernal2412 committed:
refactor: update control forms to integrate with test persistence

- Integrate embedded form with test result persistence
- Update main form to support test history features
- Improve form validation and error handling
- Enhance user experience with better feedback

alvarobernal2412 committed:
feat: enhance mashup pages with persistent test results

- Move test action to dropdown menu in Mashups list
- Integrate test history and results in ControlCreationAndTestPage
- Add test persistence across page reloads
- Improve UX with previous test access and management

alvarobernal2412 committed:
Merge branch 'main' into feature/mashup-test-button

tirirote committed:
feat: updated forms input feedback

tirirote committed:
fix: fixed input feedback for catalog wizard

tirirote committed:
chore: minor updates on the UI

tirirote committed:
fix: unified ui styles and layout

tirirote committed:
fix: minor ui bug fixe

tirirote committed:
chore: improved ui feedback

tirirote committed:
chore: reverted branch to previous commit

tirirote committed:
Merge branch 'feature/mashup-test-button' of https://github.com/statuscompliance/frontend into feature/mashup-test-button

tirirote committed:
Revert "chore: reverted branch to previous commit"

This reverts commit dc9b39f12ec6872840fcea6392c7969f96e52c16.
